### PR TITLE
fix a minor bug

### PR DESCRIPTION
**Pull request by zhenkunl**
fix: prevent gradient calculation in inference mode

Add torch.no_grad() context manager to the model inference step
to avoid unnecessary computation and memory usage.

### CODE REVIEW

File: examples/3_ResNet/resnet_infer_python.py [hunk 1]
```diff
@@ -96,6 +96,7 @@ def check_results(output: torch.Tensor) -> None:
 
     batch_size_to_run = 1
 
-    result = deploy(saved_model_file, device_to_run, batch_size_to_run)
+    with torch.no_grad():
```
Comment (jatkinson1000) [inline examples/3_ResNet/resnet_infer_python.py:99]: I'd suggest using `inference mode` as preferable to `no_grad`

```suggestion
    with torch.inference_mode():
```

**Review by jatkinson1000 — CHANGES_REQUESTED** [review]
Thanks for spotting this this @zhenkunl 

I have suggested changing for `inference_mode()` rather than `no_grad()`, though realise this would also need updating to match in the other exercises that currently use `no_grad()`.

If you can accept the suggestion I'm happy to merge this and update the others.
Alternatively, if you want to make the updates to the other exercises here we would be grateful and happily merge.
